docs/api/openapi.yaml

```diff
@@ -0,0 +1,225 @@
+        '400':
+          description: Invalid URN:NBN identifier pattern or location uri(s) supplied
+        '401':
+            $ref: '#/components/responses/UnauthorizedError'
+        '403':
+          description: URN:NBN identifier is valid, but does not match the prefix of the authenticated user
+        '409':
+          description: Conflict, resource already exists
+            
+      tags:
+        - URN:NBN identifier
+
+  /nbn/{identifier}:
+    parameters:
+    - $ref: '#/components/parameters/nbn_identifier'
+    get:
+      summary: 'Returns URN:NBN object (urn:nbn+locations) for this {identifier}.'
+      description: 'Returns URN:NBN object (urn:nbn+locations) for this {identifier}.<br>The identifier must have a prefix that matches the authenticated user. HTTP 404 otherwise.'
+      operationId: 'getNbnRecord'
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NbnLocationsObject'
+        '400':
+          description: Invalid URN:NBN identifier pattern supplied
+        '401':
+            $ref: '#/components/responses/UnauthorizedError'      
+        '404':
+          description: Supplied URN:NBN identifier not found, or URN:NBN-prefix is not registered to this user.
+      tags:
+        - URN:NBN identifier
+
+    put:
+      summary: 'Updates an existing URN:NBN {identifier} or registers a new URN:NBN {identifier} associated with a prioritized list of locations.'
+      description: 'Updates or registers a (new) URN:NBN associated with a prioritized list of locations. Multiple locations are prioritized in respective order. The first location is the preferred. The second is failover, etc. <br>Identifiers must have a prefix that matches the authenticated user.'
+      operationId: 'updateNbnRecord'
+      requestBody:
+        required: true
+        description: A json object that contains the associated locations for the {identifier}.
+        content:
+          application/json:
+            schema:
+              type: array
+              items:
+                $ref: '#/components/schemas/Location'
+      responses:
+        '200':
+          description: OK (updated existing)
+        '201':
+          description: Successful operation (created new)          
+        '400':
+          description: Invalid URN:NBN identifier or location(s) supplied
+        '401':
+            $ref: '#/components/responses/UnauthorizedError'
+      tags:
+        - URN:NBN identifier
+        
+  /nbn/{identifier}/locations:
+    parameters:
+      - $ref: '#/components/parameters/nbn_identifier'
+    get:
+      summary: 'Returns registered locations for this URN:NBN {identifier}.'
+      description: 'Returns all registed locations for this {identifier}.<br>The identifier must have a prefix that matches the authenticated user. HTTP 404 otherwise.'
```

Comment:
```suggestion
      description: 'Returns all registered locations for this {identifier}.<br>The identifier must have a prefix that matches the authenticated user. HTTP 404 otherwise.'
```